### ISSUE TITLE
Blended Diffusion for Text-driven Editing of Natural Images

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#175]
# TL; DR

- Blend noised GT image with CLIP-guided text-conditional image at each noise level during diffusion process.
-Employ multiple augmentations for CLIP gradient computation to evade adversarial training effects.

# Motivation

- GAN-based editing methods are not generic in that a real image must be well projected to the learned latent space of the generator.
- The capacity to edit image *locally* is still looked after.

# Formulation

### Local CLIP-guided diffusion

- Guide the diffusion process (classifier-guidance) using gradients from CLIP with masked images.
- Add background preservation loss (MSE + LPIPS) for isolated manipulation of masked region.
- Result: subpar trade-off between foreground alignment and background preservation.

<img width="735" alt="image" src="https://user-images.githubusercontent.com/46308407/165944561-1667944c-9cb0-4b4a-b12c-a2cdd59d1efe.png">
<img width="735" alt="image" src="https://user-images.githubusercontent.com/46308407/165944561-1667944c-9cb0-4b4a-b12c-a2cdd59d1efe.png">

### Text-driven Blended Diffusion

- Make t-th time step noised version of the input GT image (background)
- Generate t-th CLIP-guided denoised version (from Gaussian noise)
- Blend them using masked Hadamard product and use it for next round CLIP-guided diffusion.

<img width="745" alt="image" src="https://user-images.githubusercontent.com/46308407/165944766-4792dea7-9ab9-4be4-9124-c5053e576603.png">

- Note that naively optimizing for CLIP alignment with gradient-based method can result in adversarial example: although the image does not comply the text prompt, CLIP similarity is maximized.
- To evade this problem, create multiple views (augmentations) of the predicted GT image and feed to CLIP for gradient calculation.
- This way, adversarial example can still exist, but it has to fool CLIP in all augmentations: hence more difficult.

# Evaluation

<img width="586" alt="image" src="https://user-images.githubusercontent.com/46308407/165945240-5492333c-ad74-4ef2-99df-96b306bee2e3.png">
<img width="1215" alt="image" src="https://user-images.githubusercontent.com/46308407/165945220-3fbb679d-881c-4611-b42b-b9c54844ebfd.png">
<img width="1211" alt="image" src="https://user-images.githubusercontent.com/46308407/165945269-418e6fa8-3910-4b8a-8ba1-ce4ede76583a.png">
<img width="594" alt="image" src="https://user-images.githubusercontent.com/46308407/165945309-c4e62e0a-88d3-46e6-9d8e-6261b99d4b8f.png">
<img width="581" alt="image" src="https://user-images.githubusercontent.com/46308407/165945294-4bfa42e4-eb50-464e-9f70-d41eadd3e6fe.png">
<img width="1201" alt="image" src="https://user-images.githubusercontent.com/46308407/165945448-5fab538d-cca3-43cb-b0b0-1c06dc2cad56.png">
<img width="1216" alt="image" src="https://user-images.githubusercontent.com/46308407/165945494-75fc3e83-27b5-4c80-b56a-1496f97019ee.png">
<img width="586" alt="image" src="https://user-images.githubusercontent.com/46308407/165945547-85807386-055b-43e2-9475-560ca7dab50f.png">
<img width="942" alt="image" src="https://user-images.githubusercontent.com/46308407/165945627-978bc1d3-020a-405d-a1d6-2d305555928c.png">
<img width="943" alt="image" src="https://user-images.githubusercontent.com/46308407/165945601-890bd646-426f-476a-84d8-325f65170d5f.png">
![Uploading image.png…]()

# Inspiration

- Classifier guidance requires training the classifier with noised images, which is impractical with models like CLIP - hence *predict* denoised image from a noised image using the noise prediction and closed form diffusion transition.
- Analogous to image blending using Laplacian feature pyramid, blend images in all intermediate diffusion steps for coherent image editing.
- As directly optimizing CLIP gradient can lead to adversarial training, use multiple augmentations for more semantic gradient.
- Naively blending two images (foreground and background) with mask leads to incoherent output. (not spatially aligned)
- Giving scribbles to the image can provide good guidance for specific editing.
